core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -211,6 +211,11 @@ public long getSizeInBytes()
                 (rleValue != null ? rleValue.getSizeInBytes() : 0);
     }
 
```

Comment:
The same problem might occur when flattening dictionaries. Could you also address that?

Reply:
Dictionaries are trickier because in order to actually account for the "direct size", we would incur very significant overheads of tracking whether each individual position is used and what the size of each specific position is. I have an alternative strategy I think makes more sense for dictionaries but I'm hoping to open another PR for handling that since it's a slightly different approach. Basically, I think we can work around that issue by flushing dictionaries from the buffer before releasing the `PagePartitioner` back to the pool when the partitioned output operator finishes, since by definition a reused partitioner will never see the same input dictionary between uses across different drivers. I'll open that PR after this one merges since they will depend on each other at the code level though.

Reply:
Here's the follow up PR: https://github.com/trinodb/trino/pull/19806